File: 0055. Jump Game/0055.myans.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True

        far_distance = 0

        for i in range(0, len(nums)-1):
            if far_distance < i:
                return False
            
            far_distance = max(far_distance, i+nums[i])
            if far_distance >= len(nums)-1:
                return True
        
        return False

File: 0055. Jump Game/0055.myans.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut dp: Vec<bool> = vec![false; nums.len()];
        dp[0] = true;

        for i in 0..nums.len() {
            if !dp[i] {
                continue;
            }

            for j in 1..=nums[i] as usize {
                if i+j >= nums.len() {
                    break;
                }
                dp[i+j] = true;
            }
        }
        dp[nums.len()-1]
    }
}